src/encoded/static/components/cart/cart.js

```diff
@@ -1803,92 +982,137 @@ const CartComponent = ({ context, savedCartObj, inProgress, fetch, session }) =>
     const [defaultOnly, setDefaultOnly] = React.useState(true);
     // All partial file objects in the cart datasets. Not affected by currently selected facets.
     const [allFiles, setAllFiles] = React.useState([]);
+    // Tracks previous contents of `cartDatasets` to know if the cart contents have changed.
+    const cartDatasetsRef = React.useRef([]);
 
     // Retrieve current unfiltered cart information regardless of its source (object or active).
-    const { cartType, cartName, cartDatasets } = React.useMemo(() => (
-        getCartInfo(context, savedCartObj)
-    ), [context, savedCartObj]);
-
-    // Get the cart datasets subject to the dataset-type dropdown. Empty array if cart type not yet
-    // determined.
-    const cartDatasetsForType = React.useMemo(() => {
-        if (cartType) {
-            return (
-                selectedDatasetType === 'all'
-                    ? cartDatasets
-                    : cartDatasets.filter((datasetAtId) => atIdToType(datasetAtId) === selectedDatasetType)
-            );
-        }
-        return [];
-    }, [selectedDatasetType, cartDatasets, cartType]);
+    // Determine if the cart contents have changed.
+    const { cartType, cartName, cartDatasets } = getCartInfo(context, savedCartObj);
+    const isCartDatasetsChanged = !_.isEqual(cartDatasetsRef.current, cartDatasets);
```

Comment:
This is okay; but it is usually better to compare by an id rather than a deep equality